### PR TITLE
Add Kyma upgrade pipeline on kind

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added upgrade pipeline
- Added documentation
- Switched periodic job to new scripts and k8s 1.16

**Related issue(s)**

#1685

### CODE REVIEW

**Comment by michal-hudy** [comment]
This PR is removing `pre-master-kind-kubernetes-image` as it is no longer needed....

File: docs/prow/kind-jobs.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# kind Jobs
+
+This document describes how to execute kind jobs locally without Prow cluster.
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
This document describes how to trigger kind jobs locally without a Prow cluster.
```

File: docs/prow/kind-jobs.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+- [Git](https://git-scm.com/)
+- [Docker](https://www.docker.com/)
+- [Debian](https://www.debian.org/) machine
+
+## Steps
+
+### Clone repositories
+
+1. Create directory for repositories:
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
1. Create a directory for the repositories you are about to clone:
```

File: docs/prow/kind-jobs.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+   
+   ```bash
+   mkdir -p "$HOME/repositories"
+   ```
+
+2. Clone `kyma` repository:
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
2. Clone the `kyma` repository:
```

File: docs/prow/kind-jobs.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+   ```bash
+   git clone https://github.com/kyma-project/kyma.git "$HOME/repositories/kyma"
+   ```
+
+3. Clone `test-infra` repository:
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
3. Clone the `test-infra` repository:
```

File: docs/prow/kind-jobs.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+   ```bash
+   git clone https://github.com/kyma-project/test-infra.git "$HOME/repositories/test-infra"
+   ```
+
+### Start job
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
### Trigger a kind job
```

File: docs/prow/kind-jobs.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Create `artifacts` directory:
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
1. Create the `artifacts` directory:
```

File: docs/prow/kind-jobs.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+   ```bash
+   mkdir -p "$HOME/artifacts"
+   ```
+
+2. Create `docker-graph` directory:
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
2. Create the `docker-graph` directory:
```

File: docs/prow/kind-jobs.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+   ```bash
+   mkdir -p "$HOME/docker-graph"
+   ```
+
+3. Run Docker container with configuration based on ProwJob definition, for example configuration for `pre-master-kyma-kind-upgrade`:
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
3. Run a Docker container with configuration based on a ProwJob definition. For example, use the configuration for `pre-master-kyma-kind-upgrade`:
```

File: docs/prow/kind-jobs.md (hunk 1 above)
Comment (michal-hudy) [reply docs/prow/kind-jobs.md]: Hm... I'm not sure if the trigger is a good word - for me, it means something asynchronous. This instruction explains how to run a job on Docker, not Kubernetes, locally and in this scenario it is synchronous.

Comment (mmitoraj) [reply docs/prow/kind-jobs.md]: ok, so maybe `run` would be the most appropriate?
```suggestion
This document describes how to run kind jobs locally without a Prow cluster.
```

File: docs/prow/kind-jobs.md (hunk 5 above)
Comment (mmitoraj) [reply docs/prow/kind-jobs.md]: as you did not like trigger let's go with `run`:
```suggestion
### Run a kind job
```

File: prow/jobs/kyma/kyma-kind-integration.yaml [hunk 9]
```diff
@@ -118,3 +118,102 @@ presubmits:
...
+      command:
+      - /home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/kind-install-kyma.sh
+      args:
+      - --ensure-kubectl
+      - --update-hosts
+      - --delete-cluster
+      - --tune-inotify
+      - --start-docker
+      - --kubernetes-version
+      - v1.16.4
```
Comment (m00g3n) [inline prow/jobs/kyma/kyma-kind-integration.yaml]: Use 1.16.3, because there is no image 1.16.4 for kind

File: docs/prow/README.md [hunk 10]
```diff
@@ -14,6 +14,7 @@ Read the documents to learn how to:
 - [Install and manage monitoring](./prow-monitoring.md) on a Prow cluster.
 - [Create, modify, and remove component jobs using templates](./manage-component-jobs-with-templates.md) for the Prow pipeline.
 - [Update](./prow-cluster-update.md) a Prow cluster.
+- [Run kind jobs manually](./kind-jobs.md) on local environment.
```
Comment (mmitoraj) [inline docs/prow/README.md]:
```suggestion
- [Run kind jobs manually](./kind-jobs.md) in a local environment.
```

File: docs/prow/kind-jobs.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    --start-docker \
+    --kubernetes-version \
+    v1.14.6 \
+    --kyma-sources \
+    /home/prow/go/src/github.com/kyma-project/kyma \
+    --kyma-installation-timeout \
+    30m
+```
+
+All logs and artifacts generated during job run are stored in `artifacts` directory created in step 1.
```
Comment (mmitoraj) [inline docs/prow/kind-jobs.md]:
```suggestion
All logs and artifacts generated during a job run are stored in the `artifacts` directory created in step 1 of the [**Run a kind job**](#Run a kind job) section.
```

**Pull request merged**